src/control/server/mgmt_svc_test.go

```diff
@@ -130,3 +141,174 @@ func TestCheckMgmtSvcReplica(t *testing.T) {
+	})
+
+	harness := NewIOServerHarness(log)
+	harness.instances = append(harness.instances, srv)
+
+	return newMgmtSvc(harness)
+}
+
+func setupMockDrpcClientBytes(svc *mgmtSvc, respBytes []byte, err error) {
+	mi, _ := svc.harness.GetMSLeaderInstance()
+	client := &mockDrpcClient{}
+	client.setSendMsgResponse(drpc.Status_SUCCESS, respBytes)
+	client.SendMsgOutputError = err
+	mi.setDrpcClient(client)
+}
+
+func setupMockDrpcClient(svc *mgmtSvc, resp proto.Message, err error) {
+	respBytes, _ := proto.Marshal(resp)
+	setupMockDrpcClientBytes(svc, respBytes, err)
+}
+
+func newTestGetACLReq() *mgmtpb.GetACLReq {
+	return &mgmtpb.GetACLReq{
+		Uuid: "testUUID",
+	}
+}
+
+func TestPoolGetACL_NoMS(t *testing.T) {
+	log, buf := logging.NewTestLogger(t.Name())
+	defer common.ShowBufferOnFailure(t, buf)()
+
+	svc := newMgmtSvc(NewIOServerHarness(log))
+
+	resp, err := svc.PoolGetACL(nil, newTestGetACLReq())
+
+	if resp != nil {
+		t.Errorf("Expected no response, got: %+v", resp)
+	}
+
+	if err == nil {
+		t.Fatal("Expected an error, got nil")
+	}
+
+	if !strings.Contains(err.Error(), "no managed instances") {
+		t.Errorf("Expected an error about the access point, got: %v", err)
+	}
+}
+
+func TestPoolGetACL_DrpcFailed(t *testing.T) {
+	log, buf := logging.NewTestLogger(t.Name())
+	defer common.ShowBufferOnFailure(t, buf)()
+
+	svc := newTestMgmtSvc(log)
+	setupMockDrpcClient(svc, nil, errors.New("mock error"))
+
+	resp, err := svc.PoolGetACL(nil, newTestGetACLReq())
+
+	if resp != nil {
+		t.Errorf("Expected no response, got: %+v", resp)
+	}
+
+	if err == nil {
+		t.Fatal("Expected an error, got nil")
+	}
+
+	if !strings.Contains(err.Error(), "mock error") {
+		t.Errorf("Expected our mock error, got: %v", err)
+	}
+}
+
+func TestPoolGetACL_BadDrpcResp(t *testing.T) {
+	log, buf := logging.NewTestLogger(t.Name())
+	defer common.ShowBufferOnFailure(t, buf)()
+
+	svc := newTestMgmtSvc(log)
+	// dRPC call returns junk in the message body
+	badBytes := make([]byte, 12)
+	for i := range badBytes {
+		badBytes[i] = byte(i)
+	}
+
+	setupMockDrpcClientBytes(svc, badBytes, nil)
+
+	resp, err := svc.PoolGetACL(nil, newTestGetACLReq())
+
+	if resp != nil {
+		t.Errorf("Expected no response, got: %+v", resp)
+	}
+
+	if err == nil {
+		t.Fatal("Expected an error, got nil")
+	}
+
+	if !strings.Contains(err.Error(), "unmarshal") {
+		t.Errorf("Expected our mock error, got: %v", err)
+	}
+}
+
+func TestPoolGetACL_Success(t *testing.T) {
+	log, buf := logging.NewTestLogger(t.Name())
+	defer common.ShowBufferOnFailure(t, buf)()
+
+	svc := newTestMgmtSvc(log)
+
+	expectedResp := &mgmtpb.GetACLResp{
+		Status: 0,
+		ACL:    []string{"A::OWNER@:rw", "A:g:GROUP@:r"},
+	}
+	setupMockDrpcClient(svc, expectedResp, nil)
+
+	resp, err := svc.PoolGetACL(nil, newTestGetACLReq())
+
+	if err != nil {
+		t.Errorf("Expected no error, got: %v", err)
+	}
+
+	if resp == nil {
+		t.Fatal("Expected a response, got nil")
+	}
+
+	common.AssertEqual(t, resp.Status, expectedResp.Status,
```

Comment:
Could use cmp.Diff() here too:
```go
    if diff := cmp.Diff(expectedResp, resp); diff != "" {
        t.Fatalf("unexpected response: (-want, +got):\n%s\n", diff)
    }
```
This would remove the need to separately check for nil, status, and acl, with the added benefit of a nice display of the differences between what was expected vs what was received.